src/Repository/IdeaRepository.php

```diff
@@ -49,4 +50,46 @@ public function countThreadComments(Idea $idea): int
             ->getSingleScalarResult()
         ;
     }
+
+    public function countVotesByType(Idea $idea): array
+    {
+        $votes = $this
+            ->createQueryBuilder('idea')
+            ->select('vote.type, COUNT(idea) as count')
+            ->innerJoin('idea.votes', 'vote')
+            ->where('idea = :idea')
+            ->setParameter('idea', $idea)
+            ->groupBy('vote.type')
+            ->getQuery()
+            ->getArrayResult()
+        ;
+
+        return array_reduce($votes, function ($result, $vote) {
+            $result[$vote['type']] = $vote['count'];
+
+            return $result;
+        }, []);
+    }
+
+    public function countAdherentVotesForIdea(Idea $idea, Adherent $adherent): array
```

Comment:
what about `getAdherentVoteForIdea` and return Vote objects?

Reply:
Ok for renaming, but It is not in the interest to return Vote objects.